src/Command/SplitDb.php

```diff
@@ -0,0 +1,269 @@
+        ],
+        self::TYPE_SALES => [
+            DbConfig::KEY_CONNECTION => DbConfig::CONNECTION_SALES,
+            DbConfig::KEY_RESOURCE => DbConfig::RESOURCE_SALES
+        ],
+    ];
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Final database and resource configurations.
+     *
+     * @var DbConfig
+     */
+    private $dbConfig;
+
+    /**
+     * Final configuration for deploy phase
+     *
+     * @var DeployInterface
+     */
+    private $stageConfig;
+
+    /**
+     * Class for configuration merging
+     *
+     * @var ConfigMerger
+     */
+    private $configMerger;
+
+    /**
+     * Factory for creation database configurations
+     *
+     * @var RelationshipConnectionFactory
+     */
+    private $connectionDataFactory;
+
+    /**
+     * Reader for app/etc/env.php file
+     *
+     * @var ConfigReader
+     */
+    private $configReader;
+
+    /**
+     * @var MaintenanceModeSwitcher
+     */
+    private $maintenanceMode;
+
+    /**
+     *  * ./bin/magento shell wrapper.
+     *
+     * @var MagentoShell
+     */
+    private $magentoShell;
+
+    /**
+     * @param LoggerInterface $logger
+     * @param DbConfig $dbConfig
+     * @param DeployInterface $stageConfig
+     * @param ConfigMerger $configMerger
+     * @param RelationshipConnectionFactory $connectionDataFactory
+     * @param ConfigReader $configReader
+     * @param MaintenanceModeSwitcher $maintenanceModeSwitcher
+     * @param MagentoShell $magentoShell
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        DbConfig $dbConfig,
+        DeployInterface $stageConfig,
+        ConfigMerger $configMerger,
+        RelationshipConnectionFactory $connectionDataFactory,
+        ConfigReader $configReader,
+        MaintenanceModeSwitcher $maintenanceModeSwitcher,
+        MagentoShell $magentoShell
+    ) {
+        $this->logger = $logger;
+        $this->dbConfig = $dbConfig;
+        $this->stageConfig = $stageConfig;
+        $this->configMerger = $configMerger;
+        $this->connectionDataFactory = $connectionDataFactory;
+        $this->configReader = $configReader;
+        $this->maintenanceMode = $maintenanceModeSwitcher;
+        $this->magentoShell = $magentoShell;
+        parent::__construct();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function configure()
+    {
+        $types = array_keys(self::TYPE_MAP);
+        $this->setName(self::NAME)
+            ->setDescription(
+                'Move sales and/or checkout quote related tables to a separate DB servers.'
+                . ' This procedure is not allowed if you do not have prepared DBs'
+            );
+        $this->addArgument(
+            self::ARGUMENT_TYPES,
+            InputArgument::IS_ARRAY,
+            sprintf(
+                'Space-separated list of table types for migrating to appropriate DB or omit to apply to all'
```

Comment:
`Space-separated list of table types for splitting (possible types: %s). 
By default, all types of tables will be split.`
@meker12 can you please review this message?